src/charm.py

```diff
@@ -411,11 +434,20 @@ def purge_k8s_resources(self, _: RelationDepartedEvent) -> None:
     def _on_config_changed(self, _: ConfigChangedEvent) -> None:
         """Handle configuration Change."""
         if self.update_stored_state("default-storage", "default_storage_class"):
-            logger.info(
-                "Config changed. New value: 'default-storage' = '%s'",
-                self.config.get("default-storage"),
-            )
-            self.update_storage_classes()
+            default_class = self.config.get("default-storage")
```

Comment:
The `default-storage` Juju config parameter can be changed to a different default `StorageClass`. Should an action be created to list the available storage classes? I realize the `config.yaml` file already shows the available values in the description. In that case, should we validate the value here?

Reply:
Yeah, as you mention each instance in `self.resources` is more of a "handle" for the actual resource in k8s cluster.